File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        INT_MAX = 2**31 - 1
        INT_MIN = -2**31

        s = str(x)
        reversed_s = s[::-1] if x >= 0 else '-' + s[:0:-1]
        reversed_int = int(reversed_s)

        if reversed_int > INT_MAX or reversed_int < INT_MIN:
            return 0

        return reversed_int

File: 0007-reverse-integer/0007-reverse-integer.cpp
#include <iostream>
#include <limits>

class Solution {
public:
    int reverse(int x) {
        
        int reversed = 0;
        
        while (x != 0){
            int o = x % 10;
            x = x / 10;
            if (reversed > std::numeric_limits<int>::max() / 10 || (reversed == std::numeric_limits<int>::max() / 10 && o > 7)) return 0;
            if (reversed < std::numeric_limits<int>::min() / 10 || (reversed == std::numeric_limits<int>::min() / 10 && o < -8)) return 0;
            reversed = reversed* 10+ o;
        }
        
        return reversed;
    }
};